File: 1. Two Sum - Easy/twosum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
// 99.73 %  faster than online js submission
// 23.60% less mem than online js submission
 var twoSum = function(o, t) {
    const a = {};
    for(let i = 0; i < o.length; i ++) {
      if (a[t - o[i]] !== undefined) {
          return [a[t - o[i]], i]
      } else {
        a[o[i]] = i;
      }
    }
  };

File: 1. Two Sum - Easy/twosum.py
#  Faster than 96.38% other online python3 submission
#  Use less mem than 25.92% other online python3 submission
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        a = {}
        for i in range(0,len(nums)):
            if target-nums[i] in a:
                return [a[target-nums[i]],i]
            else:
                a[nums[i]]=i